guide/adopting-the-configuration-cache.md

```diff
@@ -0,0 +1,384 @@
+|         `--- :processTestResources                      (org.gradle.language.jvm.tasks.ProcessResources)
+...
+```
+
+#### Task graph of `./gradlew assemble`
+```
+:assemble                        (org.gradle.api.DefaultTask)
+`--- :jar                        (org.gradle.api.tasks.bundling.Jar)
+     +--- :classes               (org.gradle.api.DefaultTask)
+     |    +--- :compileJava      (org.gradle.api.tasks.compile.JavaCompile)
+     |    `--- :processResources (org.gradle.language.jvm.tasks.ProcessResources)
+     `--- :compileJava           (org.gradle.api.tasks.compile.JavaCompile)
+```
+
+#### Task graph of `./gradlew build`
+```
+:build                                                         (org.gradle.api.DefaultTask)
++--- :assemble                                                 (org.gradle.api.DefaultTask)
+|    `--- :jar                                                 (org.gradle.api.tasks.bundling.Jar)
+|         +--- :classes                                        (org.gradle.api.DefaultTask)
+|         |    +--- :compileJava                               (org.gradle.api.tasks.compile.JavaCompile)
+|         |    `--- :processResources                          (org.gradle.language.jvm.tasks.ProcessResources)
+|         `--- :compileJava                                    (org.gradle.api.tasks.compile.JavaCompile)
+`--- :check                                                    (org.gradle.api.DefaultTask)
+   +--- :checkstyleMain                                      (org.gradle.api.plugins.quality.Checkstyle)
+   |    +--- ...
+   +--- :checkstyleTest                                      (org.gradle.api.plugins.quality.Checkstyle)
+   |    +--- ...
+   +--- :spotlessCheck                                       (org.gradle.api.DefaultTask)
+   |    `--- ...
+   +--- :test                                                (org.gradle.api.tasks.testing.Test)
+   |    `--- ...
+   ...
+```
+
+When you first run a gradle task, the task graph is serialized and stored on disk. Upon a subsequent run of the same task, If none of these inputs have changed, Gradle skips the configuration phase entirely, loads the task graph from disk, and goes straight to the execution phase.
+
+The configuration phase typically runs faster than execution since heavy work belongs in the latter. However, without the cache, configuration phase is rerun with every single Gradle run. If you run a unit test multiple times, configuration is repeated, reproducing the same task graph every time. When the task itself is light, configuration can take up a large fraction of the latency. Configuration caching solves this by storing and reusing configuration results between runs, eliminating redundant work and speeding up iteration cycles.
+
+
+
+## Finding Configuration Cache problems
+
+To find Configuration Cache problems:
+
+1. Run `./gradlew build --configuration-cache`
```

Comment:
I think `build` is way too large a scope, and probably not that useful as a starting point. It will likely be quite overwhelming and you probably won't see results for a long time!

It's probably good as a way to maybe capture everything as lots of tasks hook on to the `build` lifecycle task transitively, but it's not a given.

I think you probably want to mention starting off with a workflow you want to "speed up" e.g. compilation workflow typically invokes: `classes` and `testClasses`, so that's what you try to find problems with.